seatunnel-connectors-v2/connector-mongodb/src/main/java/org/apache/seatunnel/connectors/seatunnel/mongodb/config/MongodbConfig.java

```diff
@@ -142,10 +142,10 @@ public class MongodbConfig {
                     .defaultValue(false)
                     .withDescription("Whether to write documents via upsert mode.");
 
-    public static final Option<List<String>> UPSERT_KEY =
-            Options.key("upsert-key")
+    public static final Option<List<String>> PRIMARY_KEY =
```

Comment:
You need add use `withFallbackKeys` to support new key and old key .
reference https://github.com/apache/seatunnel/blob/84be0f9fd057b1680d001de38802ce1c28d79f04/seatunnel-api/src/test/java/org/apache/seatunnel/api/configuration/OptionTest.java#LL49C31-L49C31